src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SuggestionModeCompletionProviderTests.cs

```diff
@@ -843,6 +843,230 @@ void A()
             await VerifyBuilderAsync(markup);
         }
 
+        [WorkItem(28586, "https://github.com/dotnet/roslyn/issues/28586")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task WithExtensionAndInstanceMethod1()
+        {
+            var markup = @"
+using System;
+
+public sealed class Foo
+{
+    public void Bar()
+    {
+    }
+}
+
+public static class FooExtensions
+{
+    public static void Bar(this Foo foo, Action<int> action)
+    {
+    }
+}
+
+public static class Repro
+{
+    public static void ReproMethod(Foo foo)
+    {
+        foo.Bar(a$$
+    }
+}
+";
+            await VerifyBuilderAsync(markup);
+        }
+
+        [WorkItem(28586, "https://github.com/dotnet/roslyn/issues/28586")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task WithExtensionAndInstanceMethod2()
+        {
+            var markup = @"
+using System;
+
+public sealed class Foo
+{
+    public void Bar()
+    {
+    }
+}
+
+public static class FooExtensions
+{
+    public static void Bar(this Foo foo, Action<int> action)
+    {
+    }
+}
+
+public static class Repro
+{
+    public static void ReproMethod(Foo foo)
+    {
+        foo.Bar(a$$)
+    }
+}
+";
+            await VerifyBuilderAsync(markup);
+        }
+
+        [WorkItem(28586, "https://github.com/dotnet/roslyn/issues/28586")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task WithExtensionAndInstanceMethod3()
+        {
+            var markup = @"
+using System;
+
+public sealed class Foo
+{
+    public void Bar()
+    {
+    }
+}
+
+public static class FooExtensions
+{
+    public static void Bar(this Foo foo, Action<int> action)
+    {
+    }
+}
+
+public static class Repro
+{
+    public static void ReproMethod(Foo foo)
+    {
+        foo.Bar(($$
```

Comment:
>( [](start = 16, length = 1)

From a quick look at `CSharpSuggestionModeCompletionProvider.IsLambdaExpression`, I would think this should have worked already (it handles a [case for open paren](https://github.com/dotnet/roslyn/blob/03548b467e6e8844db676afd6dcd616c173265b4/src/Features/CSharp/Portable/Completion/SuggestionMode/CSharpSuggestionModeCompletionProvider.cs#L123)).

Yet the bug report indicates that didn't work. What part of your change fixed this?

Reply:
That checks for specifically: ``` if (token.Parent != null && token.Parent is ParameterListSyntax)```

but the above is not a parameter list.  it's a ParenthesizedExpression.

Also, a bunch of these tests were purely to try to test *around* this feature to make sure lots of cases work properly.

Reply:
The main part of interest is: 

https://github.com/dotnet/roslyn/blob/03548b467e6e8844db676afd6dcd616c173265b4/src/Features/CSharp/Portable/Completion/SuggestionMode/CSharpSuggestionModeCompletionProvider.cs#L182-L183

We basically dont' know if this is a lambda or not (syntactically) so we use the semantic type-inference service to try to figure out.  However, that service didn't do well in this specific overload resolution case.